app/models/assignment.rb

```diff
@@ -622,4 +622,11 @@ def find_due_dates(type)
     self.due_dates.select {|due_date| due_date.deadline_type_id == DeadlineType.find_by_name(type).id }
   end
 
+  def self.first_submission_reward(assgt_id, team_id)
+    assignment = Assignment.find(assgt_id)
+    if !assignment.nil? && assignment.first_sub_teamid == -1
+      assignment.first_sub_teamid = team_id
```

Comment:
I doubt this design: you added a "first_sub_team_id" in the assignment table. This means that if we r gonna extend this badge from "first submission" to "early birds submission" to give multiple teams this badge, we have to add more than 1 column to the assignments table.

I would suggest create a badges_mapping table which records assignment_id, team_id and badge name (or badge id) in order to make this feature more extendable.